extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/db/SQLBasicSecurityStorageConnector.java

```diff
@@ -0,0 +1,1123 @@
+          public List<Map<String, Object>> inTransaction(Handle handle, TransactionStatus transactionStatus)
+              throws Exception
+          {
+            List<Map<String, Object>> role_permissions = handle
+                .createQuery(
+                    StringUtils.format(
+                        "SELECT permissions.id, permissions.resource_json\n"
+                        + "FROM permissions\n"
+                        + "WHERE permissions.role_name = :roleName"
+                    )
+                )
+                .map(permMapper)
+                .bind("roleName", roleName)
+                .list();
+            return role_permissions;
+          }
+        }
+    );
+  }
+
+  @Override
+  public List<Map<String, Object>> getPermissionsForUser(String userName)
+  {
+    return getDBI().inTransaction(
+        new TransactionCallback<List<Map<String, Object>>>()
+        {
+          @Override
+          public List<Map<String, Object>> inTransaction(Handle handle, TransactionStatus transactionStatus)
+              throws Exception
+          {
+            List<Map<String, Object>> user_permissions = handle
+                .createQuery(
+                    StringUtils.format(
+                        "SELECT permissions.id, permissions.resource_json, roles.name\n"
+                        + "FROM permissions\n"
+                        + "JOIN roles\n"
+                        + "    ON permissions.role_name = roles.name\n"
+                        + "JOIN user_roles\n"
+                        + "    ON user_roles.role_name = roles.name\n"
+                        + "WHERE user_roles.user_name = :userName"
+                    )
+                )
+                .map(permMapper)
+                .bind("userName", userName)
+                .list();
+            return user_permissions;
+          }
+        }
+    );
+  }
+
+  @Override
+  public void createAuthenticationToAuthorizationNameMapping(
+      String authenticationName, String authorizationName
+  )
+  {
+    getDBI().inTransaction(
+        new TransactionCallback<Void>()
+        {
+          @Override
+          public Void inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
+          {
+            String existingMapping = handle
+                .createQuery(
+                    StringUtils.format(
+                        "SELECT authorization_name FROM %1$s WHERE authentication_name = :authenticationName",
+                        AUTHENTICATION_AUTHORIZATION_NAME_MAPPINGS
+                    )
+                )
+                .bind("authenticationName", authenticationName)
+                .map(StringMapper.FIRST)
+                .first();
+
+            if (existingMapping == null) {
+              handle.createStatement(
+                  StringUtils.format(
+                      "INSERT INTO %1$s (authentication_name, authorization_name) VALUES (:authenticationName, :authorizationName)",
```

Comment:
Please break the line.

Reply:
This is method is deleted now since I removed the name mapping table